A basic implementation of user avatar pill

shubham-padia opened the pull request:
Fixes #9842 .

Result:
![selection_169](https://user-images.githubusercontent.com/13910561/41995964-0306498c-7a71-11e8-8867-3548434204bf.png)

User group settings:
![selection_168](https://user-images.githubusercontent.com/13910561/41995965-03681bbc-7a71-11e8-84e4-b5b611401f2f.png)

@rishig @maxnuss I've implemented the user avatar pills as above.

maxnuss commented:
Thanks @shubham-padia! I wouldn't change anything about the styling here; I think these look great. Should be ready to go!

timabbott commented:
The padding/margins at the top are a bit off on the pills (most visible if you click on them so you see the green):

![image](https://user-images.githubusercontent.com/2746074/42134653-0f4d2aea-7cf5-11e8-8213-99d285a02e1c.png)

Otherwise this lgtm.

shubham-padia commented:
> The padding/margins at the top are a bit off on the pills

@timabbott What browser are you using, I'm unable to repro this.
Chrome:
![selection_172](https://user-images.githubusercontent.com/13910561/42136689-00ebc4f0-7d7e-11e8-80e9-58e30cc9bcc0.png)

Firefox: 
![selection_171](https://user-images.githubusercontent.com/13910561/42136691-06877d5a-7d7e-11e8-9255-a1b499a3845a.png)

shubham-padia commented:
> I could also imagine rounding all four corners of the avatar?

Here's a preview:
![selection_173](https://user-images.githubusercontent.com/13910561/42136709-7552de28-7d7e-11e8-8b02-760668f4ed7e.png)

I personally like the pills without the right borders, what are your thoughts after viewing the preview ?
I'd open a feedback topic on #design otherwise.

timabbott commented:
@shubham-padia I think it's a zoom issue; I see it at 110% zoom in Chrome, but not other sizes.

maxnuss commented:
I can take a look at this too if needed

shubham-padia commented:
@timabbott Updated the PR to work for all zoom sizes.

shubham-padia commented:
Update: I've made the pill height: 20px change as a separate commit. I've added the following text to the last commit message:
```
Changes to search_pill.js are not made as search pills haven't been
added yet completely and search_pill.js just contains the preparatory
code right now.
No change to compose_pm_pill.js is not required as it uses
`user_pill.create_item_from_text` in its `create` function.
```

timabbott commented:
Nice, merged, thanks @shubham-padia!
